client.go

```diff
@@ -369,7 +353,7 @@ func (c *Client) Emit(command SocketCommand, data interface{}) error {
 	default:
 		return errors.New("command is not supported")
 	}
-	return c.ws.Emit(command, data)
+	return c.shardManager.EmitThroughAllShards(command, data)
```

Comment:
I feel the naming on these functions are a bit redundant, you're already using an instance of shardmanager, why the redundancy in the func name by mentioning shards again? I'd personally argue you'd be better off just renaming it to Emit and having a standardized Emiter interface to deal with emitting stuff.